### PR TITLE
CI: Adds auto-label Github workflow

### PR DESCRIPTION
**Pull request by sven1103**
This PR adds a new Github worfklow that labels opened or edited PRs based on their target branch name..

### CODE REVIEW

File: .github/workflows/label-pull-requests.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Label Pull Requests
+
+on:
+  pull_request:
+    types: [ opened, edited ]
```
Comment (KochTobi) [inline .github/workflows/label-pull-requests.yml:5]: this seems to be incorrect
https://renato66.github.io/auto-label/

```suggestion
    types: [ 'opened', 'edited' ]
```

**Review by KochTobi — CHANGES_REQUESTED** [review]

**Comment by KochTobi** [comment]
I think we should not depend on the branch name but rather the PR itself

**Comment by sven1103** [comment]
> I think we should not depend on the branch name but rather the PR itself

Why is that? We are using the Gitflow workflow and agreed to have a controlled vocabular for naming the branches.

File: .github/workflows/label-pull-requests.yml (hunk 1 above)
Comment (sven1103) [reply .github/workflows/label-pull-requests.yml:5]: No, not needed, these are built-in types

**Review by KochTobi — APPROVED** [review]

File: .github/workflows/label-pull-requests.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: TimonVS/pr-labeler-action@v3
+        with:
+          configuration-path: .github/pr-labels.yml # optional, .github/pr-labeler.yml is the default value
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (KochTobi) [inline .github/workflows/label-pull-requests.yml:15]: can be removed, correct?

Comment (KochTobi) [reply .github/workflows/label-pull-requests.yml:15]: wrong ... 😄

**Pull request merged**